core/web/eth_keys_controller.go

```diff
@@ -233,34 +281,40 @@ func (ekc *ETHKeysController) Chain(c *gin.Context) {
 	addressHex := c.Query("address")
 	addressBytes, err := hexutil.Decode(addressHex)
 	if err != nil {
-		jsonAPIError(c, http.StatusUnprocessableEntity, errors.Wrap(err, "invalid address"))
+		jsonAPIErrorBadRequest(c, errors.Wrap(err, "invalid address"))
 		return
 	}
 	address := common.BytesToAddress(addressBytes)
 
 	cid := c.Query("evmChainID")
 	chain, err := getChain(ekc.app.GetChains().EVM, cid)
-	if errors.Is(err, ErrInvalidChainID) || errors.Is(err, ErrMultipleChains) || errors.Is(err, ErrMissingChainID) {
-		jsonAPIError(c, http.StatusUnprocessableEntity, err)
-		return
-	} else if err != nil {
-		jsonAPIError(c, http.StatusInternalServerError, err)
-		return
+	if err != nil {
+		switch err {
+		case ErrInvalidChainID, ErrMultipleChains:
```

Comment:
It's important to maintain the `errors.Is` checks. We should probably upgrade them all.